src/invoices/InvoiceDetails/InvoiceActions/ApproveInvoiceAction/ApproveInvoiceAction.js

```diff
@@ -24,7 +25,17 @@ const ApproveInvoiceAction = ({ saveInvoice, invoice }) => {
       const approvedInvoice = { ...invoice, status: INVOICE_STATUS.approved };
 
       saveInvoice(approvedInvoice)
-        .then(() => showCallout('ui-invoice.invoice.actions.approve.success'));
+        .then(() => showCallout('ui-invoice.invoice.actions.approve.success'))
+        .catch(async (response) => {
+          try {
+            const { errors } = await response.json();
+            const errorCode = get(errors, [0, 'code']);
+
+            showCallout(`ui-invoice.invoice.actions.approve.error.${errorCode}`, 'error');
```

Comment:
what if it's a new code and there is no translation on ui yet?

Reply:
We will get toast with code for translation. 
May be I can use message from this error? Or I need every time display default message?

Reply:
don't you think that showing to user some translation keys isn't the best idea? 
it's also a case when new code introduced on the backend side 
I think you need constants for specific error codes you work in scope of this ticket, and for the rest to show general message

Reply:
You are right. I updated with constants.